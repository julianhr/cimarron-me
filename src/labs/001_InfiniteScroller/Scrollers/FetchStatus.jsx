import React from 'react'
import PropTypes from 'prop-types'
import styled from '@emotion/styled'


const Message = styled.div`
  background: #eee;
  width: 100%;
  height: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
  font-size: 20px;
  padding: 40px;
`

function FetchStatus({ children }) {
  return (
    <Message>
      <div>
        {children}
      </div>
    </Message>
  )
}

FetchStatus.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.element,
    PropTypes.string
  ]).isRequired,
}

export default FetchStatus
