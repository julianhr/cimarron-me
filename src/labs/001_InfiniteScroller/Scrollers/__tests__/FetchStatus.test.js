import React from 'react'
import { render, shallow } from 'enzyme'
import { create } from 'react-test-renderer'

import FetchStatus from '../FetchStatus'


describe('FetchStatus', () => {
  describe('props', () => {
    it('requires children', () => {
      expect(() => render(<FetchStatus />)).toThrow()
    })
  })

  it('matches snapshot with all props', () => {
    const text = 'Testing'
    const wrapper = render(<FetchStatus>{text}</FetchStatus>)
    expect(wrapper).toMatchSnapshot()
  })


  it('accepts string as child', () => {
    const text = 'Testing'
    const wrapper = render(<FetchStatus>{text}</FetchStatus>)
    expect(wrapper.text()).toEqual(text)
  })

  it('accepts element as child', () => {
    const TestElement = () => <article>testing</article>
    const wrapper = create(
      <FetchStatus>
        <TestElement className={'test-class'} />
      </FetchStatus>
    )

    expect(wrapper.root.findAllByType(TestElement)).toHaveLength(1)
    expect(wrapper).toMatchSnapshot()
  })
})
