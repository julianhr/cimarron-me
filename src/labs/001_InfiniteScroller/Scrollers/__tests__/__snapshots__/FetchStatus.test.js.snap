// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FetchStatus accepts element as child 1`] = `
<div
  className="css-quardp-Message ehwlun20"
>
  <div>
    <article>
      testing
    </article>
  </div>
</div>
`;

exports[`FetchStatus matches snapshot with all props 1`] = `
<div
  class="css-quardp-Message ehwlun20"
>
  <div>
    Testing
  </div>
</div>
`;
